fix: Initial `CommandBarFlyout` visual state

MartinZikmund opened the pull request:
**GitHub Issue:** closes https://github.com/unoplatform/uno/issues/20984

<!-- Link to relevant GitHub issue if applicable. All PRs should be associated with an issue (GitHub issue or internal), unless the change is documentation related. -->

## PR Type:🐞 Bugfix

<!--
Copy the labels that apply to this PR and paste them above:

- 🐞 Bugfix
- ✨ Feature
- 🎨 Code style update (formatting)
- 🔄 Refactoring (no functional changes, no api changes)
- 🏗️ Build or CI related changes
- 📚 Documentation content changes
- 🤖 Project automation
- 💬 Other... (Please describe)

-->


## What is the current behavior? 🤔

<!-- Please describe the current behavior that you are modifying, or link to a relevant issue. -->


## What is the new behavior? 🚀

<!-- Please describe the new behavior after your modifications. -->

## PR Checklist ✅

Please check if your PR fulfills the following requirements:

- [x] 📝 Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.
- [x] 🧪 Added [Runtime tests, UI tests, or a manual test sample](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] 📚 Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [ ] 🖼️ Validated PR `Screenshots Compare Test Run` results.
- [x] ❗ Contains **NO** breaking changes

<!-- If this PR contains a breaking change, please describe the impact and migration path for existing applications below.
     Please note that breaking changes are likely to be rejected -->

## Other information ℹ️

<!-- Please provide any additional information if necessary -->

MartinZikmund committed:
fix: Do not dispose `CompositeDisposable`

MartinZikmund committed:
fix: Correctly set `shouldShowOverflowButton`

MartinZikmund committed:
fix: Set the `IsOpen` property later

agneszitte commented on src/Uno.UI/UI/Xaml/Controls/CommandBar/CommandBar.Partial.cs:
```diff
@@ -1332,37 +1332,62 @@ private void PropagateDefaultLabelPositionToElement(ICommandBarElement element)
 			}
 		}
 
```
@MartinZikmund we would need tests as well for those changes no?

agneszitte reviewed (commented):
@MartinZikmund we would need tests as well for those changes no?

unodevops commented:
🤖 Your Docs stage site is ready! Visit it here: https://unodocsprstaging.z13.web.core.windows.net/pr-21001/docs/index.html

MartinZikmund committed:
test: Add `CommandBarFlyout` test for primary commands

This commit introduces a new test method `When_CommandBarFlyout_Without_Secondary_Commands` in the `Given_CommandBarFlyout` class. The test verifies the behavior of a `CommandBarFlyout` without secondary commands, ensuring it opens correctly, checks the visibility of primary labels, and confirms the overflow button's visibility.

unodevops commented:
🤖 Your Docs stage site is ready! Visit it here: https://unodocsprstaging.z13.web.core.windows.net/pr-21001/docs/index.html

MartinZikmund committed:
chore: Make test Uno-only

unodevops commented:
🤖 Your Docs stage site is ready! Visit it here: https://unodocsprstaging.z13.web.core.windows.net/pr-21001/docs/index.html

MartinZikmund committed:
chore: Apply suggestion from code review

Co-authored-by: Xiaotian Gu <xiaoyao312@gmail.com>

unodevops commented:
🤖 Your WebAssembly Skia Sample App stage site is ready! Visit it here: https://unowasmprstaging.z20.web.core.windows.net/pr-21001/wasm-skia-net9/index.html

MartinZikmund commented:
/azp run

unodevops commented:
🤖 Your Docs stage site is ready! Visit it here: https://unodocsprstaging.z13.web.core.windows.net/pr-21001/docs/index.html
